docs/REFERENCE.md

```diff
@@ -359,6 +408,18 @@ const fetchDataOrErrorResult = Result.from(async () => {
 });
 ```
 
+Note: If an unexpected error is thrown within the `resultFactory` function, it will be wrapped in an `UnknownError` and passed to the `mapErr` or `mapAnyErr` function if they are in the chain. If not handled, it will result in a rejected promise.
```

Comment:
```suggestion
Note: If an unexpected error is thrown within the `resultFactory` function, it will be wrapped in an `UnknownError` and passed to the `mapErr` or `mapAnyErr` function if they are in the chain. If not handled, it will result in a rejected promise when using either `.promise` or `.unsafePromise`.
```